ipn/ipnserver/server.go

```diff
@@ -1029,13 +1029,16 @@ func (s *Server) localhostHandler(ci connIdentity) http.Handler {
 // Windows and via $DEBUG_LISTENER/debug/ipn when tailscaled's --debug flag
 // is used to run a debug server.
 func (s *Server) ServeHTMLStatus(w http.ResponseWriter, r *http.Request) {
-	// As this is only meant for debug, verify there's no DNS name being used to
-	// access this.
-	if strings.IndexFunc(r.Host, unicode.IsLetter) != -1 {
+	// As this is only meant for debug, verify we are only serving to a
+	// localhost or equivalent (i.e. 127.0.0.1) origin.
+	if r.Host != "localhost" && strings.IndexFunc(r.Host, unicode.IsLetter) != -1 {
```

Comment:
it won't be "localhost". It will be "localhost:nnnnnn".  You need a SplitHostPort to check the hostname. But at that point it's worth pulling this out into a func.

Reply:
Maybe we can bring `validHost()` up a layer to ipnserver? Otherwise we could potentially get away with `strings.HasPrefix(r.Host, "localhost:")`

Reply:
strings.HasPrefix on localhost: works for now

Reply:
I ended up bringing `validHost()` a layer up.

Reply:
Update: Doing `string.HasPrefix` now as we discussed.